File: Conditionals/time.py
# Jameson Bultez, How to Get the Time of the Day

import time

# First instance of time in programming
first_time = time.gmtime # Greenwhich Mean Time

# Current time in seconds
current = time.time()

# Current date and time like we noramlly see it
now = time.ctime(current)

# Get just a part of the time
local_time = time.localtime(current)
day = local_time.tm_wday
hour = local_time.tm_hour # 24-hour time (0:00-23:59)

# Print greeting based on time of day.
if hour == 12:
    print("Good noon!")
elif 0 <= hour < 12:
    print("Good morning.")
elif 12 <= hour < 17:
    print("Good afternoon.")
elif 17 <= hour < 21:
    print("Good evening.")
else:
    print("Good night.")

File: Conditionals/time.c
// Jameson Bultez, How to Get the Time of Day

#include <stdio.h>
#include <time.h>

int main(void){
    time_t seconds; // Time in seconds since 1/1/1970 @ 0:00

    // Current time
    time_t rawtime;
    struct tm * timeinfo;

    time(&rawtime);
    timeinfo = localtime(&rawtime);

    // Current hour
    time_t now = time(NULL);

    struct tm * tm_struct = localtime(&now);
    int hour = tm_struct->tm_hour; // 24-hour time (0:00-23:59)
    
    // Print greeting based on time of day.
    if (hour == 12) {
        printf("Good noon!");
    } else if (0 <= hour < 12) {
        printf("Good morning.");
    } else if (12 <= hour < 17) {
        printf("Good afternoon.");
    } else if (17 <= hour < 21) {
        printf("Good evening.");
    } else {
        printf("Good night.");
    }
        
    return 0;
}